sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/DataSource.scala

```diff
@@ -328,6 +333,9 @@ case class DataSource(
                 "It must be specified manually")
         }
 
+        SchemaUtils.checkSchemaColumnNameDuplication(
+          dataSchema, "the datasource", sparkSession.sessionState.conf.caseSensitiveAnalysis)
```

Comment:
This branch is to load files from the metadata log. IMO, this check is not needed, unless you can reproduce the issue.